packages/protocol/src/routers/ConnextHandler.sol

```diff
@@ -216,7 +216,7 @@ contract ConnextHandler {
       revert ConnextHandler__executeFailed_tranferAlreadyExecuted(transferId, nonce);
     }
 
-    IERC20(txn.asset).approve(address(connextRouter), txn.amount);
+    SafeERC20.safeIncreaseAllowance(IERC20(txn.asset), address(connextRouter), txn.amount);
 
     try connextRouter.xBundle(actions, args) {
```

Comment:
Think the allowance should be decreased if the `xBundle(...)` call fails (assets would not have been pulled)

Reply:
Added decrease allowance in catch as suggested, considering failed execution.